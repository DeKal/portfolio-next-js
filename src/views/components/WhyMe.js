import React from 'react'
import styled from 'styled-components/macro'
import { WHYME } from '~/consts/pages'
import Title from '~/views/components/whyme/Title'
import Quote from '~/views/components/whyme/Quote'
import { Translate } from 'react-localize-redux'

const WhyMe = () => (
  <Container id={WHYME} className="container">
    <Inner className="whyme">
      <Title>
        <Translate id="Why me: Title" />
      </Title>
      <Quote>
        <Translate id="Why me: Quote" />
      </Quote>
    </Inner>
  </Container>
)

export default WhyMe

const Container = styled.section``

const Inner = styled.div`
  text-align: center;
  border-left: 10px solid ${props => props.theme.colors.colorAccent};
  border-right: 10px solid ${props => props.theme.colors.colorAccent};
  margin-top: 50px;

  @media screen and (max-width: 800px) {
    margin-top: 30px;
  }
`
