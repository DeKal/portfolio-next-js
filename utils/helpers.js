import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
