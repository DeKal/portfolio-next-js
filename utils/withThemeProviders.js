import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
