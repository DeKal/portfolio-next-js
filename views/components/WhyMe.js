import * as React from 'react'

const WhyMe = () => (
  <section id="services">
    <div className="container">
      <div className="grid-container">
        <div className="advantages">
          <h2>Why me?</h2>
          <blockquote>
            <p>
              A wide range of experiences in web framework backing up by 5 years
              of pursuing the competitive programming, I have the confidence
              that I can adapt to any languages, frameworks, or complicated
              system in a short period of time. I am a fast-learner and can be
              in charge the role of team leader when push comes to shove. On the
              other hand, I can communicate effectively in English as my
              secondary language, so the barrier of language is no longer a big
              threat. I am dedicated and will strive to make a profit for the
              team.
            </p>
          </blockquote>
        </div>
      </div>
    </div>
  </section>
)

export default WhyMe
