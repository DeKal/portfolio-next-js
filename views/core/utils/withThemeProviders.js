import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
