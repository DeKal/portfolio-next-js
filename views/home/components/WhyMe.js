import React from 'react'
import styled from 'styled-components/macro'
import Container from 'react-bootstrap/Container'
import { WHYME } from '~/consts/pages'
import Title from '~/views/home/components/whyme/Title'
import Quote from '~/views/home/components/whyme/Quote'
import { Translate } from 'react-localize-redux'

const WhyMe = () => (
  <Section id={WHYME}>
    <Inner>
      <Title>
        <Translate id="Why me: Title" />
      </Title>
      <Quote>
        <Translate id="Why me: Quote" />
      </Quote>
    </Inner>
  </Section>
)

export default WhyMe
const Section = styled(Container)`
  position: relative;
`
const Inner = styled.div`
  text-align: center;
  margin-top: 50px;

  &::after,
  &::before {
    content: '';
    position: absolute;
    top: 15px;
    height: calc(100% - 15px);
  }

  &::before {
    left: 15px;
    border-left: 10px solid ${props => props.theme.colors.colorAccent};
  }

  &::after {
    left: -15px;
    border-right: 10px solid ${props => props.theme.colors.colorAccent};
    width: 100%;
  }

  @media screen and (max-width: 800px) {
    margin-top: 30px;
  }
`
