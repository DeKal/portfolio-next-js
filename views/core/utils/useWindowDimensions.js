import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
