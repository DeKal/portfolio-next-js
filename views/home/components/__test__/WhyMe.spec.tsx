import * as React from 'react'
import { mountWithTheme } from '~/views/core/utils/withThemeProviders'
import WhyMe from '~/views/home/components/WhyMe'

describe('WhyMe', () => {

  it('should render without throwing an error', function () {
    const wrap = mountWithTheme(<WhyMe />)
    expect(wrap).toMatchSnapshot()
  })

})