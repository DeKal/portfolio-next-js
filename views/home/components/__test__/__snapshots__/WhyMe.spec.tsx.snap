// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhyMe should render without throwing an error 1`] = `
<WhyMe>
  <WhyMe__Section
    id="Whyme"
  >
    <Container
      className="WhyMe__Section-fpis95-0 bNhUkR"
      fluid={false}
      id="Whyme"
    >
      <div
        className="WhyMe__Section-fpis95-0 bNhUkR container"
        id="Whyme"
      >
        <WhyMe__Inner>
          <div
            className="WhyMe__Inner-fpis95-1 iTsQwk"
          >
            <Title>
              <Title__Container>
                <h2
                  className="Title__Container-sc-1p2849z-0 dxMJxZ"
                >
                  <Translate
                    id="Why me: Title"
                  >
                    <WrappedTranslate
                      context={
                        Object {
                          "activeLanguage": undefined,
                          "addTranslation": [Function],
                          "addTranslationForLanguage": [Function],
                          "defaultLanguage": undefined,
                          "ignoreTranslateChildren": false,
                          "initialize": [Function],
                          "languages": Array [],
                          "renderToStaticMarkup": false,
                          "setActiveLanguage": [Function],
                          "translate": [Function],
                        }
                      }
                      id="Why me: Title"
                    >
                      Missing translationId: Why me: Title for language: \${ languageCode }
                    </WrappedTranslate>
                  </Translate>
                </h2>
              </Title__Container>
            </Title>
            <Quote>
              <Quote__Container>
                <div
                  className="Quote__Container-cdfen2-0 jCZpfD"
                >
                  <Quote__Content>
                    <p
                      className="Quote__Content-cdfen2-1 iEvywS"
                    >
                      <Translate
                        id="Why me: Quote"
                      >
                        <WrappedTranslate
                          context={
                            Object {
                              "activeLanguage": undefined,
                              "addTranslation": [Function],
                              "addTranslationForLanguage": [Function],
                              "defaultLanguage": undefined,
                              "ignoreTranslateChildren": false,
                              "initialize": [Function],
                              "languages": Array [],
                              "renderToStaticMarkup": false,
                              "setActiveLanguage": [Function],
                              "translate": [Function],
                            }
                          }
                          id="Why me: Quote"
                        >
                          Missing translationId: Why me: Quote for language: \${ languageCode }
                        </WrappedTranslate>
                      </Translate>
                    </p>
                  </Quote__Content>
                </div>
              </Quote__Container>
            </Quote>
          </div>
        </WhyMe__Inner>
      </div>
    </Container>
  </WhyMe__Section>
</WhyMe>
`;
